src/main/java/com/o19s/es/ltr/feature/store/StoredFeature.java

```diff
@@ -134,7 +133,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         } else {
             builder.field(TEMPLATE.getPreferredName());
             // it's ok to use NamedXContentRegistry.EMPTY because we don't really parse we copy the structure...
-            XContentParser parser = XContentFactory.xContent(template).createParser(NamedXContentRegistry.EMPTY,
+            XContentParser parser = MediaTypeRegistry.xContent(template).xContent().createParser(NamedXContentRegistry.EMPTY,
```

Comment:
this also seems redundant (based on method names) and nests too deeply. Is there another way to not call a method called xContent twice (looking for readability). otherwise, is this something we should think about updating in core?

Reply:
I think the core is still in the middle of changes related to `XContent`, I think this is the way API is structured at the moment (but please feel free to suggest the other way of I am missing something)